stacker/config/__init__.py

```diff
@@ -106,6 +108,22 @@ def parse(raw_config):
 
     """
 
+    # Convert any applicable dictionaries back into lists
+    config_dict = yaml.safe_load(raw_config)
```

Comment:
@phobologic I was being (possibly overly) cautious about changing the loaded yaml too much, since the ordered dict loader changes every dict (top level or nested) to the OrderedDict type.

I took a pass at that now, and ended up with a bunch of test errors:
```
# Convert any applicable dictionaries back into lists
    config_dict = yaml_to_ordered_dict(raw_config)
    for top_key, top_value in config_dict.iteritems():
        if isinstance(top_value, dict) or isinstance(top_value, OrderedDict):
            tmp_list = []
            for key, value in top_value.iteritems():
                tmp_dict = dict(value)
                if top_key == 'stacks':
                    tmp_dict['name'] = key
                tmp_list.append(tmp_dict)
            config_dict[top_key] = tmp_list
```
e.g.
```
ERROR: test_stacker_build_fail_when_parameters_in_stack_def (stacker.tests.test_stacker.TestStacker)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/troyready/stacker/stacker/tests/test_stacker.py", line 107, in test_stacker_build_fail_when_parameters_in_stack_def
    stacker.configure(args)
  File "/home/troyready/stacker/stacker/commands/stacker/__init__.py", line 37, in configure
    validate=True)
  File "/home/troyready/stacker/stacker/config/__init__.py", line 51, in render_parse_load
    config = parse(rendered)
  File "/home/troyready/stacker/stacker/config/__init__.py", line 118, in parse
    tmp_dict = dict(value)
ValueError: dictionary update sequence element #0 has length 1; 2 is required
```
Possible that I'm missing something obvious in my logic here. I'm open to either way going forward.

Reply:
I'd be curious what `value` is in that case - could you add some debug statements to figure out where it is choking up?

I feel like if we're going to go with OrderedDict, it might be best to go with it through the entire config.  I may be being overly aggressive though :)

Reply:
Ok, extra weird - I took your code above (and added an import for `OrderedDict` at the top level) and ran the tests and it seemed to work fine :)

Reply:
Ahh, nm, I was testing in my stacker dir - not the one where i cloned you're repo.  Digging in now.

Reply:
I was actually able to fix most of the errors by first checking that `value` was a dict before passing it to `dict()`, just like you do with top_value (the issue was that some values were lists, so applying dict to them broke).  That said, there were a few more errors I couldn't manage to fix yesterday.  I'll try to dig into it some more later today hopefully.